ios-base/Networking/Models/APIError.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  APIError.swift
+//  ios-base
+//
+//  Created by Mauricio Cousillas on 6/4/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import Foundation
+import Moya
+
+struct APIError: Error {
+  let statusCode: Int
+  let error: RailsError
+
+  static func from(response: Response) -> APIError? {
+    guard let decodedError = try? response.map(RailsError.self) else {
+      return nil
+    }
+    return APIError(statusCode: response.statusCode, error: decodedError)
+  }
+}
+
+struct RailsError: Decodable {
+  let errors: [String: [String]]?
```

Comment:
The value under errors can also be a plain string. I think this PR may conflict with #104 which adds the base case and Error domains.

Reply:
From what I see in that PR looks like if we won't need it if we go through with Moya, since it already handles that kind of errors.
Is there any specific behavior from that PR that's missing here? Maybe I need to add some more functionality to the error handling in this PR.

Reply:
Also, we could go with custom decoding here, so we can try and parse `errors` as a dictionary, if it fails, we can try and parse it as a string and assign it to `error`

Reply:
That's a good idea. We don't want to lose that error case from the previous version and handing the entire response(`[errors: "some description"]`) to the consumer.

Reply:
Done!